simln-lib/src/lib.rs

```diff
@@ -747,18 +740,9 @@ impl Simulation {
             });
         }
 
-        // We always want to wait for all threads to exit, so we wait for all of them to exit and track any errors
```

Comment:
I don't want to lose this comment completely, so let's update it to fit the new approach.

Something like:
```We expect our producers and consumers to run until the simulation completes or we hit an error, so we can immediately close our TaskTracker and wait on existing tasks to exit.```